src/scene_logo.cpp

```diff
@@ -40,16 +40,39 @@ Scene_Logo::Scene_Logo() :
 	type = Scene::Logo;
 }
 
+Filesystem_Stream::InputStream logo_stream;
+Filesystem_Stream::InputStream next_logo;
+
 void Scene_Logo::Start() {
 	if (!Player::debug_flag && !Game_Battle::battle_test.enabled) {
 		std::time_t t = std::time(nullptr);
 		std::tm* tm = std::localtime(&t);
 
-		if (Rand::ChanceOf(1, 32) || (tm->tm_mday == 1 && tm->tm_mon == 3)) {
-			logo_img = Bitmap::Create(easyrpg_logo2, sizeof(easyrpg_logo2), false);
-		} else {
-			logo_img = Bitmap::Create(easyrpg_logo, sizeof(easyrpg_logo), false);
+		if (FileFinder::Game()) logo_stream = FileFinder::OpenImage("Font", "LOGO" + std::to_string(Player::current_logo));
```

Comment:
Using ``FileFinder::Game`` in ``Start`` has multiple problems:

1. The Game filesystem is not configured in all cases. Actually only when you use ``--project-path`` but not when you double click the Player. In that case (see ``vUpdate``) the Game FS is ``nullptr`` and it uses ``GetDefaultProjectPath``. This could be moved to ``Start`` but then there is the next problem:

2. You parse the filesystem while no logo is visible. This gives a bad user experience because reading the filesystem can be slow (that's the main reason why the heavy lifting happens in ``Update``). This means there can be seconds of black screen before the logo appears. This will be noticable on slow homebrew platforms but also on Android where parsing the filesystem can take seconds (thanks Google). In the web player you would need to wait for an entire download to finish before a logo can be displayed.

What would obviously work is that the EasyRPG logo is displayed and if there are more logos it cycles through them.

Reply:
I'm trying to change how the logos are loaded, by storing them in a Vector before start loading them:

```cpp
void Scene_Logo::LoadLogos() {
	Player::current_logo = 0;
	Player::logos_container.clear();

	bool defaultLogoLoaded = false;

	while (true) {
		Filesystem_Stream::InputStream next_logo = FileFinder::OpenImage("Font", "LOGO" + std::to_string(Player::current_logo));

		if (!next_logo) {
			// If the first logo is missing, load the default logos
			if (Player::current_logo == 0 && !defaultLogoLoaded) {
				std::time_t t = std::time(nullptr);
				std::tm* tm = std::localtime(&t);

				if (Rand::ChanceOf(1, 32) || (tm->tm_mday == 1 && tm->tm_mon == 3)) {
					Player::logos_container.push_back(Bitmap::Create(easyrpg_logo2, sizeof(easyrpg_logo2), false));
				}
				else {
					Player::logos_container.push_back(Bitmap::Create(easyrpg_logo, sizeof(easyrpg_logo), false));
				}
				defaultLogoLoaded = true;
			}
			else {
				// Break the loop if OpenImage fails for other logos
				break;
			}
		}
		else {
			Player::logos_container.push_back(next_logo);
		}

		Player::current_logo++;
	}
}
```


but not idk what exactly needs to be done, and the instructions are too complex for asking help on chatGPT.

but yeah, logo0 could always be the easyRPG one.